# <b> Membuat Analisis Data Menggunakan Jupyter Notebook (Python) </b>

<br>Nama lengkap: Fachry Dennis Heraldi
<br>NIM: 16520303
<br>Asal universitas: Institut Teknologi Bandung
<br>Tanggal mulai pengerjaan tugas: 5 November 2020
<br>Sumber dataset: TakeMeOut


# Penggunaan Library

In [7]:
import pandas as pd
from sklearn.neighbors import NearestNeighbors

import pickle

# Mulai Mengimport Data

In [27]:
df = pd.read_csv('/content/takemeout.csv')

In [28]:
df.head()

Timestamp  ... Seberapa penting bantuan dari calon pacar untuk kamu?
0  2020/10/31 3:39:25 PM GMT+7  ...                                                  3   
1  2020/10/31 3:39:36 PM GMT+7  ...                                                  2   
2  2020/10/31 3:39:38 PM GMT+7  ...                                                  4   
3  2020/10/31 3:39:42 PM GMT+7  ...                                                  3   
4  2020/10/31 3:39:43 PM GMT+7  ...                                                  4   

[5 rows x 8 columns]

# Data Preprocessing

Data cleaning

In [29]:
df = df.iloc[:, 1:]
#Menghapus kolom Timestamp karena tidak diperlukan dalam pengolahan data

In [30]:
df.head()

Siapa nama kamu?  ... Seberapa penting bantuan dari calon pacar untuk kamu?
0  A**************  ...                                                  3   
1            L****  ...                                                  2   
2     Y***********  ...                                                  4   
3             a***  ...                                                  3   
4            B****  ...                                                  4   

[5 rows x 7 columns]

In [31]:
df.columns = ['name', 'gender','qt', 'touch', 'words', 'gift','service']
#Mengubah nama kolom menjadi kata yang mudah dimengerti agar memudahkan dalam pengolahan data

In [32]:
df.head()

name gender  qt  touch  words  gift  service
0  A**************  Cowok   5      5      4     1        3
1            L****  Cewek   5      5      3     2        2
2     Y***********  Cowok   4      3      4     4        4
3             a***  Cowok   5      5      5     2        3
4            B****  Cowok   5      5      5     2        4

In [33]:
df['gender']=df['gender'].apply(lambda x: 0 if x== 'Cowok' else 1)
#Gender cewek direpresentasikan dengan 1 sedangkan cowok dengan 0

In [34]:
df.head()

name  gender  qt  touch  words  gift  service
0  A**************       0   5      5      4     1        3
1            L****       1   5      5      3     2        2
2     Y***********       0   4      3      4     4        4
3             a***       0   5      5      5     2        3
4            B****       0   5      5      5     2        4

In [35]:
df.describe()

gender          qt       touch       words        gift     service
count  101.000000  101.000000  101.000000  101.000000  101.000000  101.000000
mean     0.198020    4.099010    3.168317    3.722772    2.732673    3.752475
std      0.400495    1.212476    1.334686    1.209287    1.325829    1.143730
min      0.000000    1.000000    1.000000    1.000000    1.000000    1.000000
25%      0.000000    4.000000    2.000000    3.000000    2.000000    3.000000
50%      0.000000    5.000000    3.000000    4.000000    3.000000    4.000000
75%      0.000000    5.000000    4.000000    5.000000    4.000000    5.000000
max      1.000000    5.000000    5.000000    5.000000    5.000000    5.000000

Berdasarkan data diatas maka didapatkan bahwa partisipan mementingkan quality time dibandingkan dengan aspek yang lain. Dilihat dari rata-rata tertinggi terdapat pada qualiy time yaitu 4.099. Sedangkan gift adalah aspek yang tidak begitu dipentingkan karena memiliki nilai rata-rata terendah, yaitu hanya 2.73

In [24]:
m_df = df[df['gender'] == 0]
f_df = df[df['gender'] == 1]

In [23]:
m_x = m_df.drop(['name','gender'],axis=1)
f_x = f_df.drop(['name','gender'],axis=1)

# Model Development

Akan dicoba permodelan data dengan menyocokkan nilai aspek-aspek terdekat/mirip dengan satu partisipan dengan partisipan lainnya

In [ ]:
m_nbrs = NearestNeighbors(n_neighbors=5).fit(m_x)
f_nbrs = NearestNeighbors(n_neighbors=5).fit(f_x)

In [ ]:
X = df.drop(['name','gender'], axis=1)

In [ ]:
test = X[df['name'] == 'f*****'].values

In [ ]:
_, idx = f_nbrs.kneighbors(test)

In [ ]:
idx

array([[53, 54, 70,  3, 33]])

In [ ]:
f_df.iloc[idx[0]]['name']

67                  L*******
68                    f*****
88    u*********************
4                      B****
40                      I***
Name: name, dtype: object

In [ ]:
pickle.dump(m_nbrs, open('/content/m_model.pickle', 'wb'))
pickle.dump(f_nbrs, open('/content/f_model.pickle', 'wb'))

In [36]:
df.to_csv('/content/data.csv', index=False)
#Mengeluarkan hasil model deployment agar bisa diolah lebih lanjut dengan UI Web